src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -717,7 +766,8 @@
                             (log/trace (i18n/trs "Finish append to inventory file. ")))]
     (common/with-safe-write-lock inventory-lock inventory-lock-descriptor inventory-lock-timeout-seconds
       (log/debug (i18n/trs "Append \"{1}\" to inventory file {0}" cert-inventory entry))
-      (ks-file/atomic-write cert-inventory stream-content-fn))))
+      (ks-file/atomic-write cert-inventory stream-content-fn))
+    (maybe-write-to-infra-serial! serial-number cert-name settings)))
```

Comment:
Shouldn't we put this in the inventory-lock above?

Reply:
The serial isn't locked under any other circumstances, but it certainly wouldn't hurt.